src/main/java/com/oltpbenchmark/WorkloadState.java

```diff
@@ -56,11 +58,34 @@ public WorkloadState(BenchmarkState benchmarkState, List<Phase> works, int num_t
     }
 
     /**
-     * Add a request to do work.
+     * Add submitted procedures to the queue and wake up workers.
      */
-    public void addToQueue(int amount, boolean resetQueues) {
-        int workAdded = 0;
-        
+    public void addToQueue(List<SubmittedProcedureRun> submittedProcedureRuns) {
+        synchronized (this) {
+            int workAdded = 0;
+
+            // Add the specified number of procedures to the end of the queue.
```

Comment:
In the interest of giving you faster review feedback, and because I'm slightly knocked out from vaccine shots today, I'm punting on thinking about this. I want to be very careful in asserting that this has exactly the same behavior as the previous code. Is it possible to make `addToQueue` private?

Reply:
I needed to split `addToQueue` because I needed to make it public so that replay workloads can directly initialize and add SubmittedProcedure objects to the queue. Definitely agree on being careful about semantics.

Reply:
What do you think of just writing two separate functions, even if there is some repeated code between them? One would be what `addToQueue` is right now, and the other (called `addToQueueForCurrentPhase`) would be an exact copy of what `addToQueue` was previously.

Reply:
Actually, I realized that we may not need two functions at all. I'm tying the replay file with the Phase in the new branch, so all we need is to change `chooseTransaction` into `generateSubmittedProcedure`. The Phase will be able to either generate a random procedure or pick the next one from the replay file

Reply:
That would be preferable; an API should reflect the high-level goal when possible, and `generateSubmittedProcedure` sounds like it fits the bill.